src/v/raft/state_machine_manager.cc

```diff
@@ -218,6 +218,15 @@ ss::future<> state_machine_manager::stop() {
     co_await std::move(gate_f);
 }
 
+std::vector<state_machine_manager::entry_ptr>
+state_machine_manager::all_state_machines() const {
+    std::vector<entry_ptr> all_stms;
+    all_stms.reserve(_machines.size());
+    std::ranges::copy(
+      std::views::values(_machines), std::back_inserter(all_stms));
+    return all_stms;
+}
```

Comment:
Is it critical to copy it here, so that we snapshot the set of machines? If not, wouldn't it be neater for `do_apply_raft_snapshot` to accept a templated range argument instead?